db/warehouse/migrate/20241010005805_prune_warehouse_indexes_phase_1.rb

```diff
@@ -0,0 +1,542 @@
+# remove unused indexes from the csv loader and import tables
+class PruneWarehouseIndexesPhase1 < ActiveRecord::Migration[7.0]
+  def up
+    INDEX_TEXT.split("\n").each do |definition|
+      table, index_name, columns = parse_definition(definition)
+      remove_index_safely(table: table, index_name: index_name, columns: columns)
+    end
+  end
+
+  def down
+    # could recreate indexes with a little work. However, rebuilding might require significant IOPs so this should be
+    # done carefully
+    raise ActiveRecord::IrreversibleMigration
+  end
+
+  protected
+
+  def check_index_table(index_name, table_name)
+    sql = <<~SQL
+      SELECT t.relname
+      FROM pg_index i
+      JOIN pg_class t ON t.oid = i.indrelid
+      JOIN pg_class idx ON idx.oid = i.indexrelid
+      WHERE idx.relname = $1
+    SQL
+    result = GrdaWarehouseBase.connection.select_value(sql, 'SQL', [index_name])
+    raise "expected index #{index_name} to be on #{table_name} but was #{result}" unless result == table_name
+  end
+
+  # it seems that the index names aren't consistent between prod and dev. Attempt to remove the index by name only
+  def remove_index_safely(table:, index_name:, columns:)
```

Comment:
Another layer of safety could be to double check here that there are zero scans (and maybe idx_tup_read = 0
  and idx_tup_fetch = 0). Those two columns might always be zero if there are zero scans, so it might be redundant.